### PR TITLE
Fix over counting due to bathIR mutation when enabling cache

### PR DESCRIPTION
**Pull request by mickjermsurawong-openai**
## Summary

- Upon enabling cache, we re-use final bath IR.
- Currently, an edge case when collapsed value is null, we mutate batchIR and result in over-counting
- This PR clone the hop from IR tail so we don't mutate it.

## Test Plan

- Verified that before the fix tests failed, IR is mutated and we actually return same instance of map
- Added unit test.

## Checklist
- [ ] Documentation update

### CODE REVIEW

**Review by nikhil-zlai — APPROVED** [review]
very neat! thanks mick!

**Review by pengyu-hou — APPROVED** [review]
Thanks @mickjermsurawong-openai !

File: aggregator/src/main/scala/ai/chronon/aggregator/windowing/SawtoothMutationAggregator.scala [hunk 1]
```diff
@@ -158,7 +158,14 @@ class SawtoothMutationAggregator(aggregations: Seq[Aggregation],
           val hopIr = hopIrs(idx)
           val hopStart = hopIr.last.asInstanceOf[Long]
           if ((batchEndTs - window.millis) + tailBufferMillis > hopStart && hopStart >= queryTail) {
-            relevantHops += hopIr(baseIrIndices(i))
+            val hopVal = hopIr(baseIrIndices(i))
+            // When first relevant hop is null, there the collapsed is null,
```
Comment (pengyu-hou) [inline aggregator/src/main/scala/ai/chronon/aggregator/windowing/SawtoothMutationAggregator.scala]:
```suggestion
            // When first relevant hop is null, the collapsed is null,
```
